deploy-package/action.yml

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+name: 'package-deploy'
+description: 'Power Platform package deploy - Deploys the package dll or the zip file with a package.'
+inputs:
+  environment-url:
+    description: 'URL of Power Platform environment to connect with; e.g. "https://test-env.crm.dynamics.com"'
+    required: false
+
+  user-name:
+    description: 'Power Platform user name to authenticate with, e.g. myname@my-org.onmicrosoft.com. Setting this input makes user-name and password required; specifying alternate "app-id" credential set of inputs will result in an error.'
+    required: false
+
+  password-secret:
+    description: 'Power Platform password, required if authenticating with username. Do NOT checkin password, instead create a secret and reference it here with: see: https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#using-encrypted-secrets-in-a-workflow'
+    required: false
+
+  app-id:
+    description: 'The application id to authenticate with. Setting this input makes app-id, tenant-id and client-secret required; specifying alternate "username" credential set of inputs will result in an error.'
+    required: false
+
+  client-secret:
+    description: 'The client secret to authenticate with. Required if authenticating with app-id.'
+    required: false
+
+  tenant-id:
+    description: 'Tenant id if using app-id & client secret to authenticate.'
+    required: false
+
+  package:
+    description: 'Path to a package dll or zip file with a package.'
+    required: true
+
+  log-file:
```

Comment:
these 2 pac CLI params for logging should NOT be exposed to the action user. Instead, we should do "the right thing (TM)" which is:
we always log to console AND we set the logfile path to a known location and upload it as part of the job's artifact. There is an npm module to do so from within an action:
https://www.npmjs.com/package/@actions/artifact#Upload-an-Artifact

https://github.com/actions/toolkit/tree/main/packages/artifact